alluxio-ui/master/src/store/browse/reducer.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+import {Reducer} from 'redux';
+
+import {BrowseActionTypes, IBrowseState} from './types';
+
+const initialState: IBrowseState = {
```

Comment:
How does initial state work, would it be possible to default it without explicitly setting all the fields?

Reply:
What do you mean default it... the default state is what appears in the store before we make any api calls.

Reply:
There is a lot boilerplate here, I was wondering if there was a way to get around that.